### PR TITLE
Elasticsearch registry 

### PR DESCRIPTION
**Pull request by lapistano**
started to implement a registry using Elastica (aka elasticsearch) as backend.

### CODE REVIEW

File: Tests/Liip/Drupal/Modules/Registry/Adaptor/ElasticaAdaptorTest.php [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    {
+        $client =  new Client();
+        $index = new Index($client, self::$indexName);
+
+        if ($index->exists()) {
+
+            $response = $index->delete();
+
+            if ($response->hasError()) {
+                //$this->fail('Failed to tear down the test suite.');
```
Comment (matteodem) [inline Tests/Liip/Drupal/Modules/Registry/Adaptor/ElasticaAdaptorTest.php]: don't commit dead code

File: Tests/Liip/Drupal/Modules/Registry/Adaptor/ElasticaAdaptorTest.php [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+                print_r($response->getError());
```
Comment (matteodem) [inline Tests/Liip/Drupal/Modules/Registry/Adaptor/ElasticaAdaptorTest.php]: try to throw PHPUnit Exception

File: src/Liip/Drupal/Modules/Registry/Lucene/Elasticsearch.php [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+     * @throws \Liip\Drupal\Modules\Registry\RegistryException
+     */
+    public function unregister($identifier)
+    {
+        if (!$this->isRegistered($identifier)) {
+            throw new RegistryException(
+                $this->drupalCommonConnector->t(RegistryException::UNKNOWN_IDENTIFIER_TEXT),
+                RegistryException::UNKNOWN_IDENTIFIER_CODE
+            );
+        }
```
Comment (matteodem) [inline src/Liip/Drupal/Modules/Registry/Lucene/Elasticsearch.php:122]: abstract the if's into a  "validateRegistration" Helper function

**Comment by lapistano** [comment]
thx to @matteodem for reviewing the PR.

**Pull request merged**